examples/cluster/README.md

```diff
@@ -0,0 +1,309 @@
+
+### Step 4: Verify the Cluster
+
+Check logs:
+```bash
+kubectl -n restate logs restate-0
+```
+
+Look for messages like:
+```
+[INFO] Restate server started
+[INFO] Ingress listening on 0.0.0.0:8080
+[INFO] Admin API listening on 0.0.0.0:9070
+```
+
+## Accessing the Cluster
+
+### Port Forwarding
+
+Forward the ingress and admin ports to your local machine:
+
+```bash
+# Ingress endpoint (for invoking services)
+kubectl port-forward -n restate svc/restate 8080:8080
+
+# Admin endpoint (for deployment registration)
+kubectl port-forward -n restate svc/restate 9070:9070
+```
+
+### Testing Connectivity
+
+```bash
+# Check cluster health
+curl http://localhost:9070/health
+
+# List registered deployments
+curl http://localhost:9070/deployments
+```
+
+## Architecture
+
+### Services Created
+
+The RestateCluster creates the following Kubernetes Services:
+
+**`restate` (ClusterIP)**
+- **8080** (ingress): HTTP ingress for invoking Restate services
+- **9070** (admin): Administration API for deployment registration
+- **5122** (metrics): Metrics and inter-node communication
+
+**`restate-cluster` (Headless)**
+- **5122**: Pod-to-pod communication for clustering
+
+### Namespace and Network Policies
+
+- **Namespace**: The operator creates a dedicated `restate` namespace
+- **Network Policies**: By default, the operator configures network policies to:
+  - Allow peer-to-peer Restate traffic (port 5122)
+  - Allow operator-to-Restate admin API traffic (port 9070)
+  - Allow pods with label `allow.restate.dev/restate: "true"` to access the cluster
+  - Allow public internet egress
+  - Deny all other traffic
+
+### Storage
+
+The cluster uses a PersistentVolumeClaim for:
+- Raft log storage (metadata)
+- Local state storage
+- Partition store
+
+**Size**: 2 GiB (configurable via `spec.storage.storageRequestBytes`)
+
+## Using with Example Service
+
+The greeter service in `examples/services/greeter/` is configured to register with this cluster:
+
+```yaml
+spec:
+  restate:
+    register:
+      cluster: restate  # References this RestateCluster by name
+```
+
+Deploy the greeter service:
+```bash
+kubectl apply -f examples/services/greeter/k8s/deployment-v1.yaml
+```
+
+The operator will automatically:
+1. Register the greeter service with the Restate cluster
+2. Add the `allow.restate.dev/restate: "true"` label to greeter pods
+3. Update RestateDeployment status with deployment ID
+
+Verify registration:
+```bash
+# Port-forward admin API
+kubectl port-forward -n restate svc/restate 9070:9070
+
+# List deployments
+curl http://localhost:9070/deployments
+```
+
+## Advanced Configuration
+
+### Multi-Node Cluster
+
+For a multi-node cluster, update the manifest:
+
+```yaml
+spec:
+  compute:
+    image: restatedev/restate:1.5
+    replicas: 3  # 3 or more nodes recommended
+```
+
+### S3-Backed Metadata
+
+For production, use S3 for metadata storage:
```

Comment:
im not sure i would say this - all i say in the main readme is that its simpler to operate and i probably wouldnt want to make any stronger statement than that. raft metadata is totally fine in production